lisa/utils.py

```diff
@@ -3845,4 +3845,21 @@ def __reduce__(self):
         return (self._make_instance, (self._ctor, dct))
 
 
+class LazyMapping(Mapping):
+    """
+    Lazy Mapping dict-like class for elements evaluated on the fly.
```

Comment:
Parameters passed to the constructor need to be documented here

Reply:
We need to state that it takes the arguments in the same shape as a dict with the keys being the mapping keys and then document the value format (closures with key as param and returning the value, no automatic memoization).

Maybe stating that the parameters are the same as dict and a commented example would do it